documentation/src/main/asciidoc/topics/con_cache_store_sql.adoc

```diff
@@ -0,0 +1,22 @@
+[id='sql-cache-store_{context}']
+= SQL cache stores
+
+SQL cache stores pre-load {brandname} caches from databases, with any dialect, through JDBC connections.
+There are two types of SQL cache store:
+
+Table:: {brandname} loads entries from a single database table.
+Query:: {brandname} uses SQL queries to load entries as well as perform insert, update, and delete operations.
+
+SQL cache stores:
+
+* Allow read and write operations to persistent storage.
+* Can be read-only and act as a Cache Loader.
+* Support keys as a single column or a composite.
+* Work with simple key/value entries or with complex types defined as Protobuf schema (`.proto` files).
++
+You can generate Protobuf schema dynamically from the result set metadata or you can provide an existing schema.
```

Comment:
@ryanemerson How does Infinispan generate schema from a database? For remote caches, I presume you still need to pick up the generated schema and register them with the server. Where does Infinispan create the generated schema?